data-pipeline/src/trace_exporter.rs

```diff
@@ -197,13 +235,29 @@ impl TraceExporter {
             })
     }
 
+    /// Emit a health metric to dogstatsd
+    fn emit_metric(&self, metric: HealthMetric, custom_tags: Option<Vec<&Tag>>) {
+        let tags = match custom_tags {
```

Comment:
I don't understand anything to rust but is there a perf impact (even minimal) to create tags if we don't use them? Should this be in the `if` below?

Reply:
You're right that this can be moved into the `if` below (and I'll do so for clarity!)

From a performance perspective I'm not sure of the full effect, the variable itself can live on the stack (which will be needed no matter what), But there's likely a minor improvement here where we don't need to do a `match` and allocate an `Either` or an `iter::Chain`